HotFix.Specification/Specification.cs

```diff
@@ -47,21 +48,29 @@ public Specification Verify(Action<Session, IConfiguration> verification)
         public Specification Run()
         {
             var clock = new VirtualClock();
-            var transport = new VirtualTransport(clock, Instructions);
+            var transport = new VirtualTransport(clock, Instructions, () => LogoutRequested = true);
             var engine = new Engine { Clocks = c => clock, Transports = c => transport };
 
             var session = (Session) null;
 
             try
             {
-                session = engine.Open(Configuration);
+                using (session = engine.Open(Configuration))
+                {
+                    session.Logon();
 
-                session.Logon();
+                    while (session.Active && !LogoutRequested)
```

Comment:
As discussed, let's simplify the semantics by always calling an idempotent `Logout` method.